packages/components/src/TooltipTrigger/TooltipTrigger.component.js

```diff
@@ -22,17 +24,75 @@ const props = {
 	<Icon name="my-icon" />
 </TooltipTrigger>
  */
-function TooltipTrigger(props) {
-	const tooltip = (<Tooltip className={getTooltipClass()} id={uuid.v4()}>{props.label}</Tooltip>);
-	return (
-		<OverlayTrigger
-			placement={props.tooltipPlacement}
-			overlay={tooltip}
-			delayShow={400}
-		>
-			{props.children}
-		</OverlayTrigger>
-	);
+class TooltipTrigger extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.onMouseOver = this.onMouseOver.bind(this);
+		this.onMouseOut = this.onMouseOut.bind(this);
+		this.state = {
+			hovered: false,
+			id: uuid.v4(),
+		};
+	}
+
+	shouldComponentUpdate(nextProps, nextState) {
```

Comment:
I would have added util function which checks props difference. eg

function checkDiff(oldProps, newProps, [...propsToCheck]) {
    return !!(propsToCheck.filter(el => oldProps[el] !== newProps[el]).length);
}

and then:

shouldComponentUpdate(nextProps, nextState) {
    return checkDiff(this.props, nextProps, ['label', 'children']) 
            || checkDiff(this.state, nextState, ['hovered']);
}

but its a matter of taste